app/javascript/src/controller_pages.js

```diff
@@ -151,18 +151,29 @@ PagesController.create = function() {
 class DataController {
```

Comment:
Not recommending a change here but...

Originally, `DataController` was intended as some (generic??) component that could handle storing data on client side, updating if necessary, and then sending off to backend for permanent storage. I dare say that never happened as it was influenced by earlier expectation things might move toward frontend-app like control (which we were not keen to do for reasons of available developer resource). 

Anyway, looking through things now it definitely looks like the `DataController` name is not very descriptive as, if anything, its seems to be more of a `FormSubmitController` or `PageActionController` (or ??); whatever is appropriate and allows for reuse (I guess, if even possible). 

Doesn't matter much but if we have a current opportunity, given that I'd expect you now have a good understanding of exactly what functionality it provides, perhaps it could be renamed to be more descriptive and help with future development, etc.

Reply:
Yep, you're totally right, and I will likely be doing this in my next ticket.   I need to add (some) of the submit button functionality to forms in the settings screen, so this code is likely to be revisited then, and so I'll look at a rename then.